src/internal/modules/waiting/controller.js

```diff
@@ -1,71 +1,95 @@
-  const config = {
-    returnReminder: 'Send returns reminders',
-    returnInvitation: 'Send returns invitations'
-  };
-  return config[name];
+const getRedirectPath = (eventStatuses, ev) => {
+  if (ev.status in eventStatuses) {
+    return path.join(eventStatuses[ev.status], ev.event_id);
+  }
 };
 
-const handleProcessing = (request, h, event) => {
-  // Still processing, render the template which will refresh in 5 seconds.
+const getBillingTitle = (event, regions) => {
+  let title = getRegionName(regions, event.metadata.batch.region_id);
+  title += (event.subtype === 'two_part_tariff') ? ' two-part tariff' : ' ' + event.subtype;
+  return title + ' bill run';
+};
+
+const getWaitingForBilling = async (request, h, event) => {
+  const path = getRedirectPath({
+    'complete': '/billing/batch/summary'
+  }, event);
+  if (path) {
+    return h.redirect(path);
+  }
+  const { data } = await services.water.billingBatchCreateService.getBillingRegions();
+
   const view = {
     ...request.view,
-    pageTitle: getPageTitle(event),
-    text: 'Please wait while the mailing list is assembled. This may take a few minutes. The letters will not be sent yet.'
+    pageTitle: getBillingTitle(event, data),
+    caption: moment(event.date_created).format('D MMM YYYY'),
+    waitingType: 'billRun'
   };
+
   return h.view('nunjucks/waiting/index', view);
 };
 
-const handleReturnReminderError = (request, h) => {
-  throw Boom.badImplementation('Errored event.');
+const getWaitingForNotifications = async (request, h, event) => {
+  const path = getRedirectPath({
+    'processed': '/batch-notifications/review'
+  }, event);
+  if (path) {
+    return h.redirect(path);
+  }
+
+  const view = {
+    ...request.view,
+    pageTitle: getNotificationsTitle(event),
+    text: 'Please wait while the mailing list is assembled. This may take a few minutes. The letters will not be sent yet.',
+    waitingType: 'returns'
+  };
+
+  return h.view('nunjucks/waiting/index', view);
 };
 
-const handleReturnsRemindersProcessed = (request, h, event) => {
-  // Redirect to a new page showing a send button and a means
-  // of acquiring a csv download of all the recipients.
-  const { event_id: eventId } = event;
-  return h.redirect(`/batch-notifications/review/${eventId}`);
+const getNotificationsTitle = (ev) => {
+  const name = get(ev, 'subtype');
+  const config = {
+    returnReminder: 'Send returns reminders',
+    returnInvitation: 'Send returns invitations'
+  };
+  return config[name];
 };
 
-const subTypeHandlers = {
-  returnReminder: {
-    processing: handleProcessing,
-    processed: handleReturnsRemindersProcessed,
-    error: handleReturnReminderError
-  },
-  returnInvitation: {
-    processing: handleProcessing,
-    processed: handleReturnsRemindersProcessed,
-    error: handleReturnReminderError
-  }
+const getRegionName = (regionsArray, regionId) => {
+  const [ region ] = regionsArray.filter(region => region.regionId === regionId);
```

Comment:
`find` might be more appropriate here.

`const region = regionsArray.find(region => region.regionId === regionId);`

Reply:
done